logstash-core/lib/logstash/agent.rb

```diff
@@ -274,4 +298,13 @@ def clean_state?
   def setting(key)
     @settings.get(key)
   end
+
+  def init_pipeline_reload_metrics(id)
+    namespace = @reload_metric.namespace([id.to_sym, :reloads])
+    namespace.increment(:successes, 0)
+    namespace.increment(:failures, 0)
+    namespace.gauge(:last_error, nil)
+    namespace.gauge(:last_success_timestamp, nil)
+    namespace.gauge(:last_failure_timestamp, nil)
+  end
```

Comment:
actually I don't wish to complicate things and overengineer the metric publication, for now I'm good with tap.
I'm going to finish up the tests for this and make it a final PR